app/controllers/users.rb

```diff
@@ -0,0 +1,57 @@
+class Chitter < Sinatra::Base
+
+  get '/users/new' do
+    @user = User.new
+    erb :'users/new'
+  end
+
+  post '/users' do
+    @user = User.new(name: params[:name], username: params[:username],
+                    email: params[:email], password: params[:password],
+                    password_confirmation: params[:password_confirmation] )
+    if @user.save
+      redirect '/sessions/new'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :'users/new'
+    end
+
+  end
+
+  patch '/users' do
+    user = User.find_by_valid_token(params[:token])
+    if user.update(password: params[:password], password_confirmation: params[:password_confirmation])
```

Comment:
Metrics/LineLength: Line is too long. [102/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)